includes/class-edd-html-elements.php

```diff
@@ -510,6 +510,7 @@ public function year_dropdown( $name = 'year', $selected = 0, $years_before = 5,
 
 		$output = $this->select( array(
 			'name'             => $name,
+			'id'               => $name,
```

Comment:
I'm not 100% sure about this one, but it is breaking some tests. It would be easy enough to update the tests so they no longer expect an empty `id` attribute, which I think is the issue for at least two of the three, but just setting the `$name` to be the `id` as well may still result in duplicate IDs. Although it's not an issue currently on the exports screen so maybe this is fine. At least with the `id` set, we can associate the label with it, so that is definitely an improvement.

Reply:
Sure--so if you look at one of the [job results](https://travis-ci.org/github/easydigitaldownloads/easy-digital-downloads/jobs/737472910), you can see that two of the tests fail asserting that two strings are equal. The expected string has an empty id (`id=""`), and with the new parameter being passed, it's actually coming out as `id="year"`. [Here's the expected string in the actual test](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/issue/8112/tests/tests-html-elements.php#L138), and it should be similar for the `test_year_dropdown_variable` test. The `test_month_dropdown` is not comparing two strings, but I think it should be the same idea, changing out the empty `id` in line 186.

Reply:
I think we should add a new parameter for the ID value (same for the `month_dropdown()`). Two reasons:

1. It will allow us to set unique values (fixing the issue Robin brought up above).
2. I think `year` and `month` are too vague for us to use as ID values. We'd generally be prefixing IDs and class names with `edd-` and right now we don't have that. It seems very likely that our `year` ID could clash with another plugin or possibly even WP core somewhere.

Reply:
What do you think about `	'id'  => $form . '-' . $name,` ?  It seems a bit verbose, but I'm struggling to think of something else unique and descriptive.  @robincornett

Reply:
So it would be something like `edd_export_earnings_start_month`? (I think it needs to be all underscores as the `select` method will make any `-` into `_` in the end--double check me on that but I think that's the way it's done.) It is maybe verbose but I think it's reasonable, @LisaCee.

Reply:
I'm fine with that. :+1: